kubeadm/kube-proxy/kube-proxy.yml

```diff
@@ -0,0 +1,94 @@
+apiVersion: v1
+data:
+  run-script.ps1: |-
+    $ErrorActionPreference = "Stop";
+    mkdir -force /host/var/lib/kube-proxy/var/run/secrets/kubernetes.io/serviceaccount
+    mkdir -force /host/kube-proxy
+
+    cp -force /kube-proxy/* /host/kube-proxy
+    cp -force /var/lib/kube-proxy/* /host/var/lib/kube-proxy
+    cp -force /var/run/secrets/kubernetes.io/serviceaccount/* /host/var/lib/kube-proxy/var/run/secrets/kubernetes.io/serviceaccount #FIXME?
+
+    $networkName = (Get-Content /host/etc/cni/net.d/* | ConvertFrom-Json).name
+    $sourceVip = ($env:POD_IP -split "\.")[0..2] + 0 -join "."
+    yq w -i /host/var/lib/kube-proxy/config.conf winkernel.sourceVip $sourceVip
+    yq w -i /host/var/lib/kube-proxy/config.conf winkernel.networkName $networkName
+    yq w -i /host/var/lib/kube-proxy/config.conf featureGates.WinOverlay true
+    yq w -i /host/var/lib/kube-proxy/config.conf mode "kernelspace"
+    wins cli process run --path /kube-proxy/kube-proxy.exe --args "--v=6 --config=/var/lib/kube-proxy/config.conf --hostname-override=$env:NODE_NAME --feature-gates=WinOverlay=true"
+
+kind: ConfigMap
+apiVersion: v1
+metadata:
+  labels:
+    app: kube-proxy
+  name: kube-proxy-windows
+  namespace: kube-system
+---
+apiVersion: apps/v1
+kind: DaemonSet
+metadata:
+  labels:
+    k8s-app: kube-proxy
+  name: kube-proxy-windows
+  namespace: kube-system
+spec:
+  selector:
+    matchLabels:
+      k8s-app: kube-proxy-windows
+  template:
+    metadata:
+      labels:
+        k8s-app: kube-proxy-windows
+    spec:
+      serviceAccountName: kube-proxy
+      containers:
+      - command:
+        - powershell
+        args:
+        - -file
+        - /var/lib/kube-proxy-windows/run-script.ps1
+        env:
+        - name: NODE_NAME
+          valueFrom:
+            fieldRef:
+              apiVersion: v1
+              fieldPath: spec.nodeName
+        - name: POD_IP
+          valueFrom:
+            fieldRef:
+              fieldPath: status.podIP
+        image: sigwindowstools/kube-proxy:1.17.3
```

Comment:
ideally we should not pin the kube proxy version.
is there a way for the user to customize this during the initial script execution - e.g. using search and replace (`sed`)?

kube-proxy has a skew of +1 / -1 releases compared to core kube, so if this version is not updated and someone tries to deploy e.g. 1.19 it might fail.

Reply:
Yeah, I was not sure what to do about this problem. This manifest is not deployed/edited by a script at all, the user is doing a `kubectl apply` with this file directly. So we can document in the step where we direct the user to do this that they pick the version that matches their node/cluster. We can provide a sed command to do so, that's pretty straightforward.

Reply:
ideally it should be propagated from $KubernetesVersion in PrepareNode, so when the user picks a version the PowerShell patches the YAMLs that later the user applies.

Reply:
The problem is that `PrepareNode`/``InstallNssm` are run on the Windows nodes, whereas `kube-proxy.yml` is run from your workstation where you have `kubectl` access.

Reply:
ok, i overlooked this. i guess it makes sense to document this in that case and give a `sed` example. my guess is that we would not want the users to apply the stock manifests without making sure it has the version they want.